Modification adds resizable & repositionable window, extra hotkeys, compact view, and an alternate fuzzy searching method.

mcleodd10 opened the pull request:
This mod replaces the loop with conditional hotkeys. PgUp/PgDn will jump up/down the list 4 rows at a time, and home/end will jump to the top and bottom rows respectively. 

Pressing a number key while holding Win as a modifier will activate the corresponding window in the list.

A compact icon view can be enabled by setting 'compact' to true. The theme fonts and styles have been adjusted. The window can be clicked and dragged, or resized. The listview, input field, and window title columns will be resized relative to the window.

I've inserted a 'FuzzySearch' function, that I found somewhere online which provides better results in my experience, but I've been unsuccessful so far in finding the original author.

mcleodd10 committed:
iswitchw mod

tvjg commented:
Hi -- thanks for taking the time to submit this! Always surprising what gets traction -- two pull requests and quite a few stars in the past year. I haven't used this script in a long time as my career drifted away from the Windows space a while back. Do you know an easy way to fire up a virtual machine to test your changes? I'll try to research myself, but it may take a bit. In the meantime, I'll think about scope and how I'd like to maintain the repo going forward. Is this something you use regularly?

mcleodd10 commented:
Hey! Yeah it is something that I use myself frequently, and figured I'd try my hand at modifying your script now that I have some experience, first time contributing to a project actually, thought it'd be a good way to learn git properly 😊

I've also been able to successfully integrate listing all open Chrome tabs and selecting them from the list using g33kdudes's Chrome.ahk Library https://github.com/G33kDude/Chrome.ahk, and I'm currently looking into whether I can do the same with Firefox, since the nightly version supports the same remote debugging protocol. Initial tests work, so it's just a matter of integrating it into iswitchw.

As far as testing goes, what environment are you in currently?

mcleodd10 committed:
Add hotstrings for row select/hide when inactive

mcleodd10 committed:
Chrome integration/save window position to .ini

mcleodd10 committed:
Delete settings.ini

mcleodd10 committed:
Delete .vscode directory

mcleodd10 committed:
update gitignore

mcleodd10 committed:
more improvements

mcleodd10 committed:
Merge branch 'master' of https://github.com/evilmanimani/iswitchw into master

mcleodd10 committed:
Update README.md

mcleodd10 committed:
Update README.md

whoops

mcleodd10 committed:
Update README.md

mcleodd10 committed:
change from chrome.ahk to acc.ahk, w/ JEE funcs

mcleodd10 committed:
Update README.md

mcleodd10 committed:
Delete Chrome.ahk

mcleodd10 committed:
Condense filter method, add files/folders to list

mcleodd10 committed:
TCMatch, Vivaldi, Icon caching, basic file search

mcleodd10 committed:
remove accv2 from root, add tcmatch readme

mcleodd10 committed:
fixed typo, refresh list when removing chars

mcleodd10 committed:
Support stacked tabs in Vivaldi

mcleodd10 committed:
fix chrome tabs, comment out vivaldi support

vivaldi support is broken presently; fixed method to get tab names, but cannot select tabs currently, doDefaultAction has no effect, still figuring it out

mcleodd10 committed:
add notice re: x64 ahk

mcleodd10 committed:
whoops, fix path

mcleodd10 committed:
remove old search method, fix regex searching

Removed all remnants and related functions to the the old search method that came from the old version; TCMatch is now the only available search algorithm. It's far more developed and it makes more sense to use an existing library rather than reinventing the wheel. At this point there's very little code remaining from the earlier versions of iswitchw. Also fixed regex matching so that you can set the default search method (fuzzy search is the default) but supports the other methods as well.

mcleodd10 committed:
Update README.md

mcleodd10 commented:
At this point my version is basically a rewrite, so it wouldn't make much sense to merge them. Cheers.
